src/map/skill.cpp

```diff
@@ -1336,6 +1339,12 @@ int32 skill_additional_effect( struct block_list* src, struct block_list *bl, ui
 	if( dmg_lv < ATK_DEF ) // no damage, return;
 		return 0;
 
+	auto skill = skill_db.find(skill_id)->impl;
```

Comment:
Not really a friend of auto...
Also you need to first check if the skill was found (!= nullptr), otherwise you might crash.